### PR TITLE
[GH-1405] Add Schema for TerraDataRepoSink

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadinstitute.atlassian.net/browse/GH-1405

### CODE REVIEW

File: database/changesets/20210709_TerraDataRepoSink.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+               logicalFilePath="20210709_TerraDataRepoSink.xml"
+               runInTransaction="false">
+        <comment>
+            A descriptor for the type of TDR job wfl started while ingesting
+            outputs
+        </comment>
+        <sql>
+            CREATE TYPE TDRJobType AS ENUM (
+                'FileLoad',
+                'MetadataIngest'
```
Comment (okotsopoulos) [inline database/changesets/20210709_TerraDataRepoSink.xml:22]: Are these values also derived from TDR as like `TDRJobStatus`?

File: database/changesets/20210709_TerraDataRepoSink.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                'succeeded',
+                'failed'
+            )
+        </sql>
+        <comment>
+            The TDR Job Queue used by the `TerraDataRepo` Sink
+        </comment>
+        <sql>
+            CREATE TYPE TerraDataRepoSinkDetails AS (
+                id        BIGINT,       -- ALTER ADD GENERATED ALWAYS AS IDENTITY
```
Comment (okotsopoulos) [inline database/changesets/20210709_TerraDataRepoSink.xml:42]: (For my own understanding) this comment is flagging the alteration we'll need to run following creation from the type to get an auto-incrementing id column.

File: database/changesets/20210709_TerraDataRepoSink.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            the `TerraDataRepo` sink table schema
+        </comment>
+        <sql>
+            CREATE TABLE TerraDataRepoSink (
+                id            BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL,  -- primary key
+                dataset       TEXT NOT NULL,                                     -- the serialized EDN of the TDR dataset schema
+                dataset_table TEXT NOT NULL,                                     -- the name of the dataset table
+                from_outputs  TEXT NOT NULL,                                     -- mapping of how to coerce the output of executor to whatever the sink understands in serialized EDN
+                details       TEXT NOT NULL,                                     -- the name of the sink details table
+                CONSTRAINT TerraDataRepoSink_PKEY PRIMARY KEY (id)
```
Comment (okotsopoulos) [inline database/changesets/20210709_TerraDataRepoSink.xml:61]: Nit: other existing primary key constraints are defined in capslock, i.e. `TERRADATAREPOSINK_PKEY`.  Not sure if that was a conscious choice though.

File: database/changesets/20210709_TerraDataRepoSink.xml (hunk 1 above)
Comment (ehigham) [reply database/changesets/20210709_TerraDataRepoSink.xml:22]: Nope.

**Review by okotsopoulos — APPROVED** [review]
Looks good to me.

File: database/changesets/20210709_TerraDataRepoSink.xml (hunk 2 above)
Comment (ehigham) [reply database/changesets/20210709_TerraDataRepoSink.xml:42]: yeah exactly!

File: database/changesets/20210709_TerraDataRepoSink.xml (hunk 3 above)
Comment (ehigham) [reply database/changesets/20210709_TerraDataRepoSink.xml:61]: Thanks - it's good to be consistent!

**Pull request merged**